js/apps/admin-ui/src/identity-providers/add/ExtendedOAuth2Settings.tsx

```diff
@@ -0,0 +1,119 @@
+};
+
+export const ExtendedOAuth2Settings = () => {
+  const { t } = useTranslation();
+  const { control } = useFormContext();
+
+  const [isExpanded, setIsExpanded] = useState(false);
+  const [promptOpen, setPromptOpen] = useState(false);
+
+  return (
+    <ExpandableSection
+      toggleText={t("advanced")}
+      onToggle={() => setIsExpanded(!isExpanded)}
+      isExpanded={isExpanded}
+    >
+      <Form isHorizontal>
+        <TextField field="config.defaultScope" label="scopes" />
+        <FormGroupField label="prompt">
+          <Controller
+            name="config.prompt"
+            defaultValue=""
+            control={control}
+            render={({ field }) => (
+              <KeycloakSelect
+                toggleId="prompt"
+                onToggle={() => setPromptOpen(!promptOpen)}
+                onSelect={(value) => {
+                  field.onChange(value as string);
+                  setPromptOpen(false);
+                }}
+                selections={field.value || t(`prompts.unspecified`)}
+                variant={SelectVariant.single}
+                aria-label={t("prompt")}
+                isOpen={promptOpen}
+              >
+                {Object.entries(promptOptions).map(([key, val]) => (
+                  <SelectOption
+                    selected={val === field.value}
+                    key={key}
+                    value={val}
+                  >
+                    {t(`prompts.${key}`)}
+                  </SelectOption>
+                ))}
+              </KeycloakSelect>
+            )}
+          />
+        </FormGroupField>
+        <SwitchField
+          field="config.acceptsPromptNoneForwardFromClient"
+          label="acceptsPromptNone"
+        />
+        <FormGroup
+          label={t("allowedClockSkew")}
+          labelIcon={
+            <HelpItem
+              helpText={t("allowedClockSkewHelp")}
+              fieldLabelId="allowedClockSkew"
+            />
+          }
+          fieldId="allowedClockSkew"
+        >
+          <Controller
+            name="config.allowedClockSkew"
+            defaultValue={0}
+            control={control}
+            render={({ field }) => {
+              const v = Number(field.value);
+              return (
+                <NumberInput
+                  data-testid="allowedClockSkew"
+                  inputName="allowedClockSkew"
+                  min={0}
+                  max={2147483}
+                  value={v}
+                  readOnly
+                  onPlus={() => field.onChange(v + 1)}
+                  onMinus={() => field.onChange(v - 1)}
+                  onChange={(event) => {
+                    const value = Number(
+                      (event.target as HTMLInputElement).value,
+                    );
+                    field.onChange(value < 0 ? 0 : value);
+                  }}
+                />
+              );
+            }}
+          />
+        </FormGroup>
```

Comment:
It would be better to use a `NumberControl` here
```suggestion
       <NumberControl
            name="config.allowedClockSkew"
            label={t("allowedClockSkew")}
            labelIcon={t("allowedClockSkewHelp")}
            controller={{ defaultValue: 0, rules: { min: 0 } }}
          />
```